_posts/2024-12-18-Management-Console-on-OpenShift.adoc

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title:  "Using the management console on OpenShift"
+date:   2024-12-18
+tags:   hal management console openshift
+author: hpehl
+description: How to use the management console for WildFly instances running on OpenShift.
+---
+
+In this blog post I'd like to show how you can use the management console (aka https://hal.github.io[HAL]) for WildFly instances running on OpenShift.
+
+== Prerequisites
+
+The console is an integral part of WildFly and is activated by default when running on-premise. For instances running on OpenShift, the console is not available by default, though. To use the console on OpenShift, you need a WildFly image that meets the following requirements:
+
+* Management user: +
+  The management console is protected by default and requires a management user added with the `add-user.sh` script.
+* Public route to the management interface: +
+  The management interface has to be publicly available from the outside.
+* Allowed origin: +
+  The console uses the https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API[fetch API] to talk to the management interface of a running WildFly instance. In an OpenShift environment, the origins of the public route and the management interface itself are different. That's why we need to tell WildFly that it is ok to make requests to the management interface from another origin (see https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS[CORS policies] for more details).
+
+You can build such an image on your own based on the official WildFly images available at https://quay.io/repository/wildfly/wildfly[quay.io/wildfly/wildfly] (see "Extending the image"). Another way is to use the pre-built images from https://quay.io/repository/halconsole/wildfly[quay.io/halconsole/wildfly]. These images are mainly meant for HAL development and testing but already meet these requirements, which makes them suitable for our use case. In particular the images add a management user `admin:admin` and have a list of preconfigured allowed origins.
+
```

Comment:
Perhaps a WARNING admonition block saying this preconfigured admin:admin setup is for development purposes or something? I'd be concerned if people start running this in production and relying only on allowed-origins to control access.